### ISSUE TITLE
Error "got exception for batch of"

### ISSUE THREAD

**Issue by engenegr** [ACINQ/eclair#2260]
Error, initially discovered on Testnet node

```
2022-05-05 16:39:36,157 ERROR f.a.e.b.b.r.BatchingClient - got exception for batch of 49 requestssttp.client3.SttpClientException$ConnectException: Exception when sending request: POST http://192.168.2.6:18332/wallet/%2Fhome%2Fu%2F.bitcoin%2Ftestnet3%2Feclair
	at sttp.client3.SttpClientException$.defaultExceptionToSttpClientException(SttpClientException.scala:40)
	at sttp.client3.okhttp.OkHttpBackend$.exceptionToSttpClientException(OkHttpBackend.scala:175)
	at sttp.client3.okhttp.OkHttpBackend.$anonfun$adjustExceptions$1(OkHttpBackend.scala:50)
	at sttp.client3.SttpClientException$$anonfun$adjustExceptions$1.applyOrElse(SttpClientException.scala:59)
	at sttp.client3.SttpClientException$$anonfun$adjustExceptions$1.applyOrElse(SttpClientException.scala:58)
	at scala.concurrent.impl.Promise$Transformation.run(Promise.scala:490)
	at java.base/java.util.concurrent.ForkJoinTask$RunnableExecuteAction.exec(Unknown Source)
	at java.base/java.util.concurrent.ForkJoinTask.doExec(Unknown Source)
	at java.base/java.util.concurrent.ForkJoinPool$WorkQueue.topLevelExec(Unknown Source)
	at java.base/java.util.concurrent.ForkJoinPool.scan(Unknown Source)
	at java.base/java.util.concurrent.ForkJoinPool.runWorker(Unknown Source)
	at java.base/java.util.concurrent.ForkJoinWorkerThread.run(Unknown Source)
Caused by: java.net.ProtocolException: unexpected end of stream
	at okhttp3.internal.http1.Http1ExchangeCodec$FixedLengthSource.read(Http1ExchangeCodec.kt:371)
	at okhttp3.internal.connection.Exchange$ResponseBodySource.read(Exchange.kt:276)
	at okio.RealBufferedSource$inputStream$1.read(RealBufferedSource.kt:158)
	at sttp.client3.internal.package$.transfer$1(package.scala:23)
	at sttp.client3.internal.package$.transfer(package.scala:30)
	at sttp.client3.internal.package$.toByteArray(package.scala:35)
	at sttp.client3.okhttp.BodyFromOkHttp$$anon$1.$anonfun$regularAsByteArray$1(BodyFromOkHttp.scala:61)
	at scala.util.Try$.apply(Try.scala:210)
	at sttp.client3.okhttp.BodyFromOkHttp$$anon$1.regularAsByteArray(BodyFromOkHttp.scala:61)
	at sttp.client3.okhttp.BodyFromOkHttp$$anon$1.regularAsByteArray(BodyFromOkHttp.scala:46)
	at sttp.client3.internal.BodyFromResponseAs.$anonfun$doApply$13(BodyFromResponseAs.scala:46)
	at sttp.monad.syntax$MonadErrorOps.map(MonadError.scala:60)
	at sttp.client3.internal.BodyFromResponseAs.doApply(BodyFromResponseAs.scala:46)
	at sttp.client3.internal.BodyFromResponseAs.$anonfun$doApply$1(BodyFromResponseAs.scala:24)
	at sttp.monad.syntax$MonadErrorOps.map(MonadError.scala:60)
	at sttp.client3.internal.BodyFromResponseAs.doApply(BodyFromResponseAs.scala:24)
	at sttp.client3.internal.BodyFromResponseAs.$anonfun$apply$1(BodyFromResponseAs.scala:14)
	at sttp.monad.syntax$MonadErrorOps.map(MonadError.scala:60)
	at sttp.client3.internal.BodyFromResponseAs.apply(BodyFromResponseAs.scala:14)
	at sttp.client3.okhttp.BodyFromOkHttp.apply(BodyFromOkHttp.scala:43)
	at sttp.client3.okhttp.BodyFromOkHttp.apply$(BodyFromOkHttp.scala:38)
	at sttp.client3.okhttp.OkHttpFutureBackend$$anon$1.apply(OkHttpFutureBackend.scala:30)
	at sttp.client3.okhttp.OkHttpBackend.readResponse(OkHttpBackend.scala:95)
	at sttp.client3.okhttp.OkHttpAsyncBackend$$anon$1.onResponse(OkHttpAsyncBackend.scala:37)
```

I expect that Eclair should work without errors.

I've seen that on Eclair Testnet and on Mainnet as well.

Compiled, 0.7.0-a804905.

Server with Linux OS installed.

Both nodes work with remote Core in VPN networks.

Dump from Mainnet node

```
2022-05-05 11:47:58,951 ERROR f.a.e.b.b.r.BatchingClient - got exception for batch of 1 requestsjava.net.ProtocolException: unexpected end of stream
2022-05-05 11:48:29,360 ERROR f.a.e.b.b.r.BatchingClient - got exception for batch of 2 requestsjava.net.ProtocolException: unexpected end of stream
2022-05-05 11:54:10,786 ERROR f.a.e.b.b.r.BatchingClient - got exception for batch of 1 requestsjava.net.ProtocolException: unexpected end of stream
2022-05-05 11:54:42,047 ERROR f.a.e.b.b.r.BatchingClient - got exception for batch of 2 requestsjava.net.ProtocolException: unexpected end of stream
2022-05-05 12:02:05,089 ERROR f.a.e.b.b.r.BatchingClient - got exception for batch of 1 requestsjava.net.ProtocolException: unexpected end of stream
2022-05-05 12:02:35,897 ERROR f.a.e.b.b.r.BatchingClient - got exception for batch of 2 requestsjava.net.ProtocolException: unexpected end of stream
2022-05-05 12:06:04,547 ERROR f.a.e.b.b.r.BatchingClient - got exception for batch of 1 requestsjava.net.ProtocolException: unexpected end of stream
2022-05-05 12:06:35,604 ERROR f.a.e.b.b.r.BatchingClient - got exception for batch of 2 requestsjava.net.ProtocolException: unexpected end of stream
2022-05-05 12:20:26,511 ERROR f.a.e.b.b.r.BatchingClient - got exception for batch of 1 requestsjava.net.ProtocolException: unexpected end of stream
2022-05-05 12:20:57,508 ERROR f.a.e.b.b.r.BatchingClient - got exception for batch of 2 requestsjava.net.ProtocolException: unexpected end of stream
2022-05-05 12:24:10,407 ERROR f.a.e.b.b.r.BatchingClient - got exception for batch of 1 requestsjava.net.ProtocolException: unexpected end of stream
2022-05-05 12:24:43,978 ERROR f.a.e.b.b.r.BatchingClient - got exception for batch of 2 requestsjava.net.ProtocolException: unexpected end of stream
2022-05-05 12:29:12,764 ERROR f.a.e.b.b.r.BatchingClient - got exception for batch of 1 requestsjava.net.ProtocolException: unexpected end of stream
2022-05-05 12:29:43,447 ERROR f.a.e.b.b.r.BatchingClient - got exception for batch of 2 requestsjava.net.ProtocolException: unexpected end of stream
2022-05-05 12:41:32,647 ERROR f.a.e.b.b.r.BatchingClient - got exception for batch of 1 requestsjava.net.ProtocolException: unexpected end of stream
2022-05-05 12:56:57,563 ERROR f.a.e.b.b.r.BatchingClient - got exception for batch of 1 requestsjava.net.ProtocolException: unexpected end of stream
2022-05-05 12:57:28,284 ERROR f.a.e.b.b.r.BatchingClient - got exception for batch of 2 requestsjava.net.ProtocolException: unexpected end of stream
2022-05-05 13:17:40,822 ERROR f.a.e.b.b.r.BatchingClient - got exception for batch of 1 requestsjava.net.ProtocolException: unexpected end of stream
2022-05-05 13:18:11,569 ERROR f.a.e.b.b.r.BatchingClient - got exception for batch of 2 requestsjava.net.ProtocolException: unexpected end of stream
2022-05-05 13:19:12,828 ERROR f.a.e.b.b.r.BatchingClient - got exception for batch of 1 requestsjava.net.ProtocolException: unexpected end of stream
2022-05-05 13:19:43,812 ERROR f.a.e.b.b.r.BatchingClient - got exception for batch of 2 requestsjava.net.ProtocolException: unexpected end of stream
2022-05-05 13:22:19,646 ERROR f.a.e.b.b.r.BatchingClient - got exception for batch of 1 requestsjava.net.ProtocolException: unexpected end of stream
2022-05-05 13:22:50,225 ERROR f.a.e.b.b.r.BatchingClient - got exception for batch of 2 requestsjava.net.ProtocolException: unexpected end of stream
2022-05-05 13:35:53,811 ERROR f.a.e.b.b.r.BatchingClient - got exception for batch of 1 requestsjava.net.ProtocolException: unexpected end of stream
2022-05-05 13:58:08,800 ERROR f.a.e.b.b.r.BatchingClient - got exception for batch of 1 requestsjava.net.ProtocolException: unexpected end of stream
2022-05-05 13:58:39,335 ERROR f.a.e.b.b.r.BatchingClient - got exception for batch of 5 requestsjava.net.ProtocolException: unexpected end of stream
2022-05-05 14:25:47,923 ERROR f.a.e.b.b.r.BatchingClient - got exception for batch of 1 requestsjava.net.ProtocolException: unexpected end of stream
2022-05-05 14:40:29,768 ERROR f.a.e.b.b.r.BatchingClient - got exception for batch of 1 requestsjava.net.ProtocolException: unexpected end of stream
2022-05-05 14:41:28,221 ERROR f.a.e.b.b.r.BatchingClient - got exception for batch of 3 requestsjava.net.ProtocolException: unexpected end of stream
2022-05-05 14:50:56,540 ERROR f.a.e.b.b.r.BatchingClient - got exception for batch of 1 requestsjava.net.ProtocolException: unexpected end of stream
2022-05-05 15:04:16,580 ERROR f.a.e.b.b.r.BatchingClient - got exception for batch of 1 requestsjava.net.ProtocolException: unexpected end of stream
2022-05-05 15:04:47,032 ERROR f.a.e.b.b.r.BatchingClient - got exception for batch of 2 requestsjava.net.ProtocolException: unexpected end of stream
2022-05-05 15:05:40,275 ERROR f.a.e.b.b.r.BatchingClient - got exception for batch of 1 requestsjava.net.ProtocolException: unexpected end of stream
2022-05-05 15:06:11,357 ERROR f.a.e.b.b.r.BatchingClient - got exception for batch of 2 requestsjava.net.ProtocolException: unexpected end of stream
2022-05-05 15:06:41,895 ERROR f.a.e.b.b.r.BatchingClient - got exception for batch of 3 requestsjava.net.ProtocolException: unexpected end of stream
2022-05-05 15:19:37,862 ERROR f.a.e.b.b.r.BatchingClient - got exception for batch of 1 requestsjava.net.ProtocolException: unexpected end of stream
2022-05-05 15:32:08,473 ERROR f.a.e.b.b.r.BatchingClient - got exception for batch of 1 requestsjava.net.ProtocolException: unexpected end of stream
2022-05-05 15:32:39,383 ERROR f.a.e.b.b.r.BatchingClient - got exception for batch of 2 requestsjava.net.ProtocolException: unexpected end of stream
2022-05-05 16:16:50,978 ERROR f.a.e.b.b.r.BatchingClient - got exception for batch of 1 requestsjava.net.ProtocolException: unexpected end of stream
2022-05-05 16:17:21,621 ERROR f.a.e.b.b.r.BatchingClient - got exception for batch of 2 requestsjava.net.ProtocolException: unexpected end of stream
2022-05-05 16:17:51,926 ERROR f.a.e.b.b.r.BatchingClient - got exception for batch of 3 requestsjava.net.ProtocolException: unexpected end of stream
2022-05-05 16:41:45,361 ERROR f.a.e.b.b.r.BatchingClient - got exception for batch of 1 requestsjava.net.ProtocolException: unexpected end of stream
```

**Comment by t-bast** [comment]
Well, what's the issue exactly? Are you seeing anything not functioning correctly? This log line indicates that a network call to `bitcoind` failed. There is nothing eclair can do about it: your network seems to be unreliable, that's completely outside of eclair's realm. We will keep logging those, because they mean you should somehow fix your network reliability to avoid disturbances.

Regardless of these logs, eclair does handle these errors gracefully (by retrying or aborting some sub-protocol if necessary). Are you seeing a behavior that indicates otherwise?

**Comment by t-bast** [comment]
> messages on Testnet are unlikely to be really large or numerous and I got this error exactly when node started.

You're confused. This has nothing to do with messages on testnet. The messages we're dealing with here are RPC requests/responses between eclair and bitcoind. There can be many messages flowing between eclair and bitcoind for channel management, unrelated to whether you're using testnet or mainnet.

Again, this is a network error: this happens outside of eclair. The only thing eclair can do are:

- gracefully handle these errors -> it should be the case, if you see a behavior that indicates otherwise then it's worth pointing out
- reduce the likelihood that these errors happen: if you're able to identify why your network is failing for some requests, then maybe we can tweak eclair to be "nicer" to the network conditions

But eclair fundamentally cannot prevent these errors from happening if they're due to an unreliable network.

**Comment by t-bast** [comment]
@engenegr were you able to test this?

**Comment by engenegr** [comment]
@t-bast yes, I was able to test it but I was distracted by other issues in our setup and didn't pay much attention specifically to batching issues. I will continue and look into it soon.

**Comment by engenegr** [comment]
Opened PR.

**Comment by t-bast** [comment]
Fixed by #2301
